index.js

```diff
@@ -66,6 +66,10 @@ module.exports = function (val, opts, pad) {
 			objKeys = Object.keys(val);
 
 			var ret = '{\n' + objKeys.map(function (el, i) {
+				if (opts.filter && !opts.filter.call(this, val, el)) {
```

Comment:
> Why are you using .call?

no special reason to use `.call`, it can just `opts.filter(val, el)`.

> `val` should probably be `val[el]`

`el` provides a way to check the property name, what is not possible passing `val[el]` param. For example:

my object:

```
{
   prop1: 1,
   prop2: 2,
   prop3: {
      prop4: 4
   }
}
```

my filter:

```
filter: function (val, el) {
	return el !== 'prop1';
}
```

passing `val[el]` instead of `val` and `el`, it wouldn't be possible to filter by 'prop1', but only second-level elements.